src/rez/package_order.py

```diff
@@ -608,6 +609,265 @@ def from_pod(cls, data):
+                if default_key != -1:
+                    raise ValueError("version_priorities may only have one "
+                                     "False / empty value")
+                default_key = priority_sort_key
+                continue
+            if version_range.contains_version(version):
+                break
+        else:
+            # For now, we're permissive with the version_sort_order - it may
+            # contain ranges which match no actual versions, and if an actual
+            # version matches no entry in the version_sort_order, it is simply
+            # placed after other entries
+            priority_sort_key = default_key
+
+        if self.version_orderer:
+            version_key = self.version_orderer.sort_key_implementation(package_name, version)
+        else:
+            version_key = version
+
+        return priority_sort_key, version_key
+
+    @classmethod
+    def _packages_to_pod(cls, packages: Dict[str, List[Union[str, VersionRange]]]):
+        return {
+            package: [str(v) for v in versions]
+            for (package, versions) in packages.items()
+        }
+
+    @classmethod
+    def _packages_from_pod(cls, packages: Dict[str, List[Union[str, VersionRange]]]):
+        from rez.version import VersionRange
+        parsed_dict = {}
+        for package, versions in packages.items():
+            new_versions = []
+            num_false = 0
+            for v in versions:
+                if v in ("", False):
+                    v = False
+                    num_false += 1
+                else:
+                    if not isinstance(v, VersionRange):
+                        if isinstance(v, (int, float)):
+                            v = str(v)
+                        v = VersionRange(v)
+                new_versions.append(v)
+            if num_false > 1:
+                raise ConfigurationError("version_priorities for CustomPackageOrder may "
+                                         "only have one False / empty value")
+            parsed_dict[package] = new_versions
+        return parsed_dict
+
+    def to_pod(self):
+        return dict(packages=self._packages_to_pod(self.packages_dict))
+
+    @classmethod
+    def from_pod(cls, data):
+        return cls(packages=data["packages"])
+
+
+class PyPAPackageOrder(PackageOrder):
+    """A package order that allows for package ordering according to PEP440/PyPA.
+
+    A prerelease argument can be specified to order prerelease versions in front
+    of release versions
+
+    For example, given the versions [1.0b2, 1.0a1, 1.0, 1.1, 1.0rc1, 1.1b2],
+    an orderer initialized with ``prerelease=""`` would give the order:
+     [1.1, 1.1b2, 1.0, 1.0rc1, 1.0b2, 1.0a1].
+    an orderer initialized with ``prerelease="b"`` would give the order:
+     [1.1b2, 1.1, 1.0b2, 1.0rc1, 1.0, 1.0a1].
+
+    """
+    name = "pypa"
+
+    def __init__(self, prerelease: Literal["", "rc", "b", "a"] = "", packages=None):
```

Comment:
it could be a nice feature to allow "beta" and "alpha" as valid input options, and to map them to "b" and "a" for the value of `self.prerelease`. 

you an make a type variable to hold and reuse this literal at the top of the module:

```python
PrereleaseType: TypeAlias = Literal["rc", "b", "beta", "a", "alpha"]
```

Reply:
From the pep440 spec: "Pre-releases allow the additional spellings of alpha, beta, c, pre, and preview for a, b, rc, rc, and rc respectively."